File: backend/app/schemas/scenario.py
"""Scenario request/response schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ScenarioCreate(BaseModel):
    """Create a new scenario."""

    name: str
    hcp_profile_id: str
    rubric_id: str
    description: str = ""
    tags: list[str] = []
    mode: str = "f2f"
    difficulty: str = "medium"
    status: str = "draft"
    key_messages: list[str] = []
    skill_id: str | None = None
    pass_threshold: int = 70


class ScenarioUpdate(BaseModel):
    """Update an existing scenario. All fields optional for partial updates."""

    name: str | None = None
    hcp_profile_id: str | None = None
    rubric_id: str | None = None
    description: str | None = None
    tags: list[str] | None = None
    mode: str | None = None
    difficulty: str | None = None
    status: str | None = None
    key_messages: list[str] | None = None
    skill_id: str | None = None
    pass_threshold: int | None = None


class ScenarioResponse(BaseModel):
    """Scenario response with all fields."""

    id: str
    name: str
    description: str
    tags: str  # JSON string from DB
    mode: str
    difficulty: str
    status: str
    hcp_profile_id: str
    key_messages: str  # JSON string from DB
    skill_id: str | None = None
    skill_version_id: str | None = None
    rubric_id: str
    pass_threshold: int
    created_by: str
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/scenario.ts
import type { HcpProfile } from "./hcp";

export interface Scenario {
  id: string;
  name: string;
  description: string;
  tags: string[];
  mode: "f2f" | "conference";
  difficulty: "easy" | "medium" | "hard";
  status: "draft" | "active" | "archived";
  hcp_profile_id: string;
  hcp_profile?: HcpProfile;
  key_messages: string[];
  skill_id: string;
  skill_version_id: string | null;
  rubric_id: string;
  pass_threshold: number;
  created_by: string;
  created_at: string;
  updated_at: string;
}

export interface ScenarioCreate {
  name: string;
  tags?: string[];
  hcp_profile_id: string;
  skill_id: string;
  rubric_id: string;
  description?: string;
  mode?: Scenario["mode"];
  difficulty?: Scenario["difficulty"];
  key_messages?: string[];
  pass_threshold?: number;
}

export interface ScenarioUpdate {
  name?: string;
  tags?: string[];
  hcp_profile_id?: string;
  skill_id?: string;
  rubric_id?: string;
  description?: string;
  mode?: Scenario["mode"];
  difficulty?: Scenario["difficulty"];
  key_messages?: string[];
  pass_threshold?: number;
}
